src/main/kotlin/io/emeraldpay/dshackle/upstream/UpstreamRpcModulesDetector.kt

```diff
@@ -0,0 +1,39 @@
+package io.emeraldpay.dshackle.upstream
+
+import com.fasterxml.jackson.core.type.TypeReference
+import io.emeraldpay.dshackle.Global
+import io.emeraldpay.dshackle.upstream.rpcclient.ListParams
+import org.slf4j.LoggerFactory
+import reactor.core.publisher.Mono
+
+
+abstract class UpstreamRpcModulesDetector(
```

Comment:
Should be in chain specific (cause BasicUpstreamRpcModulesDetector suitable only for eth like networks)